From faa8ecb8a6f444f62ce1e3c093e3727138c02b56 Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/installer/pull/14647
From: Tom Deseyn <tom.deseyn@gmail.com>
Date: Tue, 4 Oct 2022 08:58:58 +0200
Subject: [PATCH] BundledVersions: update portable rid logic.

Defaults the portable rid to match the rid.
This works for portable builds.

For non-portable builds, the portable rid can be
controlled using PortableOSName.

For source-build, we set PortableOSName based
on the bootstrap SDK's portable rid.
---
 .../tarball/content/repos/installer.proj         |  5 +++++
 src/redist/targets/GetRuntimeInformation.targets | 16 +++++++++-------
 2 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/repos/installer.proj b/repos/installer.proj
index f6803f4cf64..4226bc62be3 100644
--- a/repos/installer.proj
+++ b/repos/installer.proj
@@ -10,6 +10,10 @@
     <OverrideTargetRid Condition="'$(TargetOS)' == 'OSX'">osx-x64</OverrideTargetRid>
     <OSNameOverride>$(OverrideTargetRid.Substring(0, $(OverrideTargetRid.IndexOf("-"))))</OSNameOverride>
 
+    <!-- Determine target portable rid based on bootstrap SDK's portable rid -->
+    <_platformIndex>$(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-'))</_platformIndex>
+    <PortableOS Condition="'$(PortableOS)' == ''">$(NETCoreSdkPortableRuntimeIdentifier.Substring(0, $(_platformIndex)))</PortableOS>
+
     <RuntimeArg>--runtime-id $(OverrideTargetRid)</RuntimeArg>
     <RuntimeArg Condition="'$(TargetOS)' == 'Linux'">--runtime-id $(TargetRid)</RuntimeArg>
 
@@ -22,6 +26,7 @@
     -->
     <BuildCommandArgs>$(BuildCommandArgs) /p:NETCoreAppMaximumVersion=99.9</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:OSName=$(OSNameOverride)</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) /p:PortableOSName=$(PortableOS)</BuildCommandArgs>
     <BuildCommandArgs Condition="'$(TargetOS)' == 'Linux'">$(BuildCommandArgs) /p:Rid=$(TargetRid)</BuildCommandArgs>
     <BuildCommandArgs>$(BuildCommandArgs) /p:DOTNET_INSTALL_DIR=$(DotNetCliToolDir)</BuildCommandArgs>
 
